io/zenoh-transport/src/multicast/establishment.rs

```diff
@@ -11,75 +11,81 @@
-    //         ConduitSn {
-    //             reliable: prng.gen_range(0..manager.config.sn_resolution),
-    //             best_effort: prng.gen_range(0..manager.config.sn_resolution),
-    //         }
-    //     };
-    // }
+    // Generate initial SNs
+    let sn_resolution = manager.config.resolution.get(Field::FrameSN);
+    let max = seq_num::get_mask(sn_resolution);
+    macro_rules! zgen_prioritysn {
+        () => {
+            PrioritySn {
+                reliable: prng.gen_range(0..=max),
+                best_effort: prng.gen_range(0..=max),
+            }
+        };
+    }
 
-    // let locator = link.get_dst();
-    // let initial_sns = if manager.config.multicast.is_qos {
-    //     let mut initial_sns = [ConduitSn::default(); Priority::NUM];
-    //     for isn in initial_sns.iter_mut() {
-    //         *isn = zgen_conduit_sn!();
-    //     }
-    //     ConduitSnList::QoS(initial_sns.into())
-    // } else {
-    //     ConduitSnList::Plain(zgen_conduit_sn!())
-    // };
-    // let config = TransportMulticastConfig {
-    //     manager: manager.clone(),
-    //     initial_sns,
-    //     link: link.clone(),
-    // };
-    // let ti = Arc::new(TransportMulticastInner::make(config)?);
+    let initial_sns = if manager.config.multicast.is_qos {
+        (0..Priority::NUM).map(|_| zgen_prioritysn!()).collect()
+    } else {
+        vec![zgen_prioritysn!()]
+    }
+    .into_boxed_slice();
 
-    // // Store the active transport
-    // let transport: TransportMulticast = (&ti).into();
-    // zlock!(manager.state.multicast.transports).insert(locator.to_owned(), ti.clone());
+    // Create the transport
+    let locator = link.get_dst().to_owned();
+    let config = TransportConfigMulticast {
+        link,
+        sn_resolution,
+        initial_sns,
+        #[cfg(feature = "shared-memory")]
+        is_shm: manager.config.multicast.is_shm,
+    };
+    let ti = Arc::new(TransportMulticastInner::make(manager.clone(), config)?);
 
-    // // Notify the transport event handler
-    // let batch_size = manager.config.batch_size.min(link.get_mtu());
-    // ti.start_tx(batch_size).map_err(|e| {
-    //     zlock!(manager.state.multicast.transports).remove(locator);
-    //     let _ = ti.stop_tx();
-    //     e
-    // })?;
-    // let callback = manager
-    //     .config
-    //     .handler
-    //     .new_multicast(transport.clone())
-    //     .map_err(|e| {
-    //         zlock!(manager.state.multicast.transports).remove(locator);
-    //         let _ = ti.stop_tx();
-    //         e
-    //     })?;
-    // ti.set_callback(callback);
-    // ti.start_rx().map_err(|e| {
-    //     zlock!(manager.state.multicast.transports).remove(locator);
-    //     let _ = ti.stop_rx();
-    //     e
-    // })?;
+    // Add the link
+    ti.start_tx()?;
 
-    // Ok(transport)
+    // Store the active transport
+    zasynclock!(manager.state.multicast.transports).insert(locator.clone(), ti.clone());
```

Comment:
What if there's alerady a transport at this locator? (insert would return `Some(previous_ti)`)